contracts/test/integration/IntegrationTestPCVGuardian.t.sol

```diff
@@ -0,0 +1,215 @@
+
+        pcvGuardian = new PCVGuardian(
+            address(core),
+            address(this), // using 'this' address as the safe address for withdrawals
+            whitelistAddresses
+        );
+
+        pcvGuardAdmin = new PCVGuardAdmin(address(core));
+
+        // grant the pcvGuardian the PCV controller and Guardian roles
+        vm.startPrank(addresses.voltDeployerAddress);
+        core.grantPCVController(address(pcvGuardian));
+        core.grantGuardian(address(pcvGuardian));
+
+        // create the PCV_GUARD_ADMIN role and grant it to the PCVGuardAdmin contract
+        core.createRole(TribeRoles.PCV_GUARD_ADMIN, TribeRoles.GOVERNOR);
+        core.grantRole(TribeRoles.PCV_GUARD_ADMIN, address(pcvGuardAdmin));
+
+        // create the PCV guard role, and grant it to the 'guard' address
+        core.createRole(TribeRoles.PCV_GUARD, TribeRoles.PCV_GUARD_ADMIN);
+        pcvGuardAdmin.grantPCVGuardRole(guard);
+        vm.stopPrank();
+    }
+
+    function testPCVGuardianRoles() public {
+        assertTrue(core.isGuardian(address(pcvGuardian)));
+        assertTrue(core.isPCVController(address(pcvGuardian)));
+    }
+
+    function testPCVGuardAdminRole() public {
+        assertTrue(
+            core.hasRole(TribeRoles.PCV_GUARD_ADMIN, address(pcvGuardAdmin))
+        );
+    }
+
+    function testGovernorWithdrawToSafeAddress() public {
+        vm.startPrank(addresses.voltDeployerAddress);
+        assertEq(fei.balanceOf(address(this)), 0);
+
+        pcvGuardian.withdrawToSafeAddress(address(pcvDeposit), withdrawAmount);
+
+        assertEq(fei.balanceOf(address(this)), withdrawAmount);
+    }
+
+    function testGovernorWithdrawAllToSafeAddress() public {
+        vm.startPrank(addresses.voltDeployerAddress);
+        assertEq(fei.balanceOf(address(this)), 0);
+
+        uint256 amountToWithdraw = pcvDeposit.balance();
+        pcvGuardian.withdrawAllToSafeAddress(address(pcvDeposit));
+
+        assertEq(fei.balanceOf(address(this)), amountToWithdraw);
+    }
+
+    function testGuardianWithdrawToSafeAddress() public {
+        vm.prank(addresses.voltDeployerAddress);
+        core.grantGuardian(address(0x1234));
+
+        assertEq(fei.balanceOf(address(this)), 0);
+
+        vm.prank(address(0x1234));
+        pcvGuardian.withdrawToSafeAddress(address(pcvDeposit), withdrawAmount);
+
+        assertEq(fei.balanceOf(address(this)), withdrawAmount);
+    }
+
+    function testGuardianWithdrawAllToSafeAddress() public {
+        vm.prank(addresses.voltDeployerAddress);
+        core.grantGuardian(address(0x1234));
+
+        assertEq(fei.balanceOf(address(this)), 0);
+        uint256 amountToWithdraw = pcvDeposit.balance();
+
+        vm.prank(address(0x1234));
+        pcvGuardian.withdrawAllToSafeAddress(address(pcvDeposit));
+
+        assertEq(fei.balanceOf(address(this)), amountToWithdraw);
+    }
+
+    function testGuardWithdrawToSafeAddress() public {
+        vm.startPrank(guard);
```

Comment:
if you startPrank, I believe you are supposed to stopPrank, though not sure this is required

Reply:
Each test runs in it's own little 'silo', so in this case doesn't really make a difference, I believe it would though if for example we didn't stopPrank() in the setup, or if we were emulating multiple addresses with multiple calls in the same test case

Reply:
Definitely best practice to explicitly clean up after the prank and call `stopPrank()` though